frontend/js/behaviors/core/dynamicFilter.js

```diff
@@ -5,79 +5,86 @@ import getStringSimilarity from '../../functions/core/getStringSimilarity';
 
 const dynamicFilter = function(container) {
   const listingContainer = container.querySelector('[data-filter-target]');
-  const listingItems = listingContainer ? listingContainer.querySelectorAll('.m-listing') : [];
+  let listingItems = listingContainer ? listingContainer.querySelectorAll('.m-listing') : [];
 
   let setup = false; // State tracking of if setup is done so we don't reflow calling and looping the components
   let registeredParameters = []; // Parameters pulled from components to evaluate and update
-  
+  let noResultsElement = null; // Element to show when no results are found
+  let castParameters = [];
+
+  // Helper function to refresh the current listingItems
+  function refreshListingItems() {
+    if (listingContainer) {
+      listingItems = listingContainer.querySelectorAll('.m-listing');
+    }
+  }
+
   function initFromUrl() {
     const url = new URL(window.location.href);
-    let castParameters = [];
-    
-    console.log('init from url');
-    console.log(registeredParameters);
-
     registeredParameters.forEach(item => {
       const paramValue = url.searchParams.get(item.parameter);
 
-      if (paramValue !== null && (item.value.includes(paramValue) || item.value === "")) {
+      if (paramValue !== null || paramValue == "" || typeof paramValue == "integer") {
```

Comment:
SonarQube raised an issue with this line:
https://sonarcloud.io/project/issues?pullRequest=745&open=AZYDafDRTaqzFr39XvDB&id=art-institute-of-chicago_artic.edu

I think it should be `"number"` instead of `"integer"`:
https://sonarcloud.io/organizations/artinstituteofchicago-github/rules?open=javascript%3AS4125&rule_key=javascript%3AS4125